// @barabel324/react-player/dist/index-DsPXrYFA.js
import { jsx as d } from "react/jsx-runtime";
import { useRef as a, useImperativeHandle as l, useEffect as u } from "react";
import { u as p } from "./index-CbgYp99p.js";
const m = async () => new Promise((t, n) => {
  if (window.VK?.VideoPlayer) {
    t(window.VK);
    return;
  }
  const o = "https://vk.com/js/api/videoplayer.js";
  if (Array.from(document.scripts).find(
    (r) => r.src === o
  )) {
    const r = setInterval(() => {
      window.VK?.VideoPlayer && (clearInterval(r), t(window.VK));
    }, 300);
    setTimeout(() => {
      clearInterval(r), n(new Error("VK Video API timeout"));
    }, 15e3);
    return;
  }
  const e = document.createElement("script");
  e.src = o, e.async = !0, e.onload = () => {
    const r = () => {
      window.VK?.VideoPlayer ? t(window.VK) : setTimeout(r, 300);
    };
    r();
  }, e.onerror = () => {
    n(new Error("Failed to load VK Video API"));
  };
  const c = document.getElementsByTagName("script")[0];
  c.parentNode?.insertBefore(e, c);
}), f = (t) => {
  const n = /^https:\/\/vkvideo\.ru\/video(-\d+)_(\d+)$/, o = t.match(n);
  if (!o)
    throw new Error(`Некорректный формат ссылки ${t}`);
  const i = o[1], e = o[2];
  return `https://vkvideo.ru/video_ext.php?oid=${i}&id=${e}`;
}, h = ({
  className: t,
  ref: n,
  url: o
}) => {
  const i = a(null), e = a(null), c = p(), r = () => {
    e.current && e.current.play?.();
  };
  return l(n, () => ({
    play: r
  })), u(() => {
    if (i.current)
      return m().then((s) => {
        e.current = s.VideoPlayer(i.current), c({ type: "canplay" });
      }).catch((s) => {
        console.error(s);
      }), () => {
        e.current && e.current.destroy();
      };
  }, []), /* @__PURE__ */ d(
    "iframe",
    {
      className: t,
      width: "100%",
      height: "100%",
      ref: i,
      src: `${f(o)}&js_api=1`,
      allow: "autoplay; fullscreen; accelerometer; gyroscope; picture-in-picture; encrypted-media",
      "data-testid": "embed-iframe",
      frameBorder: "0",
      scrolling: "no",
      title: "player",
      allowFullScreen: !0
    }
  );
};
export {
  h as default
};
