// @barabel324/react-player/dist/index-CLC2KZWC.js
import { jsx as l } from "react/jsx-runtime";
import { useRef as s, useImperativeHandle as p, useEffect as y } from "react";
import { u as f } from "./index-CbgYp99p.js";
const m = (t) => {
  const n = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/, o = t.match(n);
  return o && o[7].length === 11 ? o[7] : "";
}, w = async () => await new Promise((t) => {
  if (window.YT)
    t(window.YT);
  else {
    const n = "https://www.youtube.com/iframe_api", r = Array.from(document.querySelectorAll("script")).find((e) => e.src === n) ?? document.createElement("script");
    if (r.src) {
      const e = setInterval(() => {
        window.YT?.PlayerState && (t(window.YT), clearInterval(e));
      }, 300);
      setTimeout(() => {
        clearInterval(e);
      }, 6e4);
    } else {
      r.src = n;
      const e = document.getElementsByTagName("script")[0];
      e.parentNode && e.parentNode.insertBefore(r, e), window.onYouTubeIframeAPIReady = () => {
        t(window.YT);
      };
    }
  }
}), Y = ({
  className: t,
  ref: n,
  url: o
}) => {
  const r = s(null), e = s(null), i = f(), u = (a) => {
    e.current = a.target, i({
      type: "canplay"
    });
  }, d = () => {
    e.current?.playVideo && e.current.playVideo();
  };
  return p(n, () => ({
    play: d
  })), y(() => {
    if (!r.current) return;
    const a = m(o);
    w().then((c) => {
      new c.Player(r.current, {
        width: "100%",
        height: "100%",
        videoId: a,
        events: {
          onReady: u
        }
      });
    }).catch((c) => {
      console.error(c);
    });
  }, []), /* @__PURE__ */ l("div", { className: t, ref: r });
};
export {
  Y as default
};
