// @barabel324/react-player/dist/index-CbgYp99p.js
import { jsx as a, jsxs as d } from "react/jsx-runtime";
import { createContext as b, useContext as m, lazy as _, useReducer as g, useMemo as k, useRef as q } from "react";
function v() {
  let e, r = "", t = 0;
  for (; t < arguments.length; ) (e = arguments[t++]) && typeof e == "string" && (r && (r += " "), r += e);
  return r;
}
const E = "_button_1xali_2", R = {
  button: E
}, S = ({
  className: e,
  onClick: r
}) => /* @__PURE__ */ a(
  "button",
  {
    className: v(
      R.button,
      e
    ),
    type: "button",
    onClick: r
  }
), f = {
  showButton: !1,
  showPreview: !0,
  canplay: !1
}, B = b(f), x = b(() => {
}), T = () => {
  const e = m(B);
  if (e === void 0)
    throw new Error("usePlayerState нужно использовать внутри PlayerProvider");
  return e;
}, C = () => {
  const e = m(x);
  if (e === void 0)
    throw new Error("usePlayerDispatch нужно использовать внутри PlayerProvider");
  return e;
}, c = {
  show: "show",
  hide: "hide",
  showButton: "showButton",
  hideButton: "hideButton",
  canplay: "canplay"
}, $ = "_playerPreview_lqhcw_1", D = "_playerPreview__wrapper_lqhcw_7", V = "_playerPreview__wrapper_cursor_lqhcw_11", j = "_playerPreview__button_lqhcw_14", A = "_playerPreview__preview_lqhcw_21", L = "_playerPreview__player_lqhcw_30", o = {
  playerPreview: $,
  playerPreview__wrapper: D,
  playerPreview__wrapper_cursor: V,
  playerPreview__button: j,
  playerPreview__preview: A,
  playerPreview__player: L
}, O = ({
  className: e,
  preview: r,
  onPlay: t,
  children: u,
  buttonComponent: n,
  previewPictureComponent: l
}) => {
  const { showButton: i, showPreview: p, canplay: P } = T(), y = C(), h = () => {
    P && (y({
      type: c.hide
    }), t && t());
  }, w = !!r || !!l;
  return /* @__PURE__ */ d(
    "div",
    {
      className: v(
        o.playerPreview,
        e
      ),
      children: [
        (i || w && p) && /* @__PURE__ */ d(
          "div",
          {
            className: v(
              o.playerPreview__wrapper,
              i && o.playerPreview__wrapper_cursor
            ),
            onClick: h,
            children: [
              i && (n ? /* @__PURE__ */ a(n.component, { className: o.playerPreview__button, ...n.props }) : /* @__PURE__ */ a(
                S,
                {
                  className: o.playerPreview__button
                }
              )),
              w && p && (l ? /* @__PURE__ */ a(l.component, { className: o.playerPreview__preview, ...l.props }) : r && /* @__PURE__ */ a("img", { className: o.playerPreview__preview, src: r.src, alt: r.alt ?? "preview" }))
            ]
          }
        ),
        /* @__PURE__ */ a("div", { className: o.playerPreview__player, children: u })
      ]
    }
  );
}, s = {
  youtube: "youtube",
  rutube: "rutube",
  video: "video",
  vk: "vk"
}, Y = {
  [s.youtube]: _(async () => await import("./index-CLC2KZWC.js")),
  [s.rutube]: _(async () => await import("./index-DKL3JxMY.js")),
  [s.video]: _(async () => await import("./index-hAoMgMeI.js")),
  [s.vk]: _(async () => await import("./index-DsPXrYFA.js"))
}, z = (e) => {
  const r = /^(?:https?:\/\/)?(?:m\.|www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/, t = /^(?:https?:\/\/)?(?:rutu\.be\/|rutube\.ru(\/play\/))/, u = /^(?:https?:\/\/)?(?:www\.|m\.)?(?:vk\.com|vkvideo\.ru)\/.+$/;
  return t.test(e) ? s.rutube : r.test(e) ? s.youtube : u.test(e) ? s.vk : s.video;
}, I = "_player_1afml_1", M = {
  player: I
}, U = (e, r) => {
  const { type: t } = r;
  switch (t) {
    case c.show:
      return { ...e, showButton: !0, showPreview: !0 };
    case c.hide:
      return { ...e, showButton: !1, showPreview: !1 };
    case c.showButton:
      return { ...e, showButton: !0 };
    case c.hideButton:
      return { ...e, showButton: !1 };
    case c.canplay:
      return { ...e, canplay: !0, showButton: !0 };
    default:
      return e;
  }
}, F = ({ children: e }) => {
  const [r, t] = g(U, f);
  return /* @__PURE__ */ a(B.Provider, { value: r, children: /* @__PURE__ */ a(x.Provider, { value: t, children: e }) });
}, G = ({
  children: e,
  hasPreview: r,
  ...t
}) => r ? /* @__PURE__ */ a(
  O,
  {
    ...t,
    children: e
  }
) : e, H = ({
  className: e,
  url: r,
  preview: t,
  classNames: u,
  components: n
}) => {
  const l = u?.player, i = n?.button, p = n?.previewPicture, P = k(() => {
    const N = z(r);
    return Y[N];
  }, [r]), y = q(null), h = () => {
    y.current && y.current.play();
  }, w = !!t || !!p;
  return /* @__PURE__ */ a(
    G,
    {
      hasPreview: w,
      className: e,
      preview: t,
      onPlay: h,
      buttonComponent: i,
      previewPictureComponent: p,
      children: /* @__PURE__ */ a(
        P,
        {
          className: v(
            l ?? M.player,
            !w && e
          ),
          ref: y,
          url: r
        }
      )
    }
  );
}, Q = (e) => /* @__PURE__ */ a(F, { children: /* @__PURE__ */ a(H, { ...e }) });
export {
  c as P,
  Q as a,
  C as u
};
